src/Microsoft.DotNet.ProjectModel/OutputPathCalculator.cs

```diff
@@ -10,7 +11,10 @@ public class OutputPathCalculator
     {
         private const string ObjDirectoryName = "obj";
 
-        private readonly ProjectContext _project;
+        private readonly Project _project;
```

Comment:
Yep, wrong interpretation. The context is more than that and potentially includes a runtime ID,a  lock file, etc. For project references there's no need to create a context to get the output paths. You only need reference information.